enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/config/EnforcerConfig.java

```diff
@@ -161,13 +161,18 @@ public JWTTransformer getJwtTransformer(String issuer) {
         if (jwtTransformerMap.containsKey(issuer)) {
             return jwtTransformerMap.get(issuer);
         }
-        JWTTransformer defaultJWTTransformer = new DefaultJWTTransformer();
-        jwtTransformerMap.put(issuer, defaultJWTTransformer);
-        return defaultJWTTransformer;
+        synchronized (jwtTransformerMap) {
+            if (jwtTransformerMap.containsKey(issuer)) {
```

Comment:
We still have the same set of lines in 161

Reply:
Yeah, for line 161 thread is not blocked, if the thread enters here we check again if the transformer exists. If two thread wait here and one update the map, so the second one should check again and create a new one if not exists

Reply:
Understood. Shall we add a comment?